pkg/storegateway/bucket_test.go

```diff
@@ -2652,14 +2688,56 @@ func TestBlockSeries_Cache(t *testing.T) {
 		// We break the index cache to not allow looking up series, so we know we don't look up series.
 		indexr.block.indexCache = forbiddenFetchMultiSeriesForRefsIndexCache{b.indexCache, t}
 		for i, tc := range testCases {
-			ss, _, err := blockSeries(context.Background(), indexr, nil, nil, tc.matchers, tc.shard, cachedSeriesHasher{shc}, nil, sl, true, b.meta.MinTime, b.meta.MaxTime, log.NewNopLogger())
+			ss, _, err := blockSeries(context.Background(), indexr, nil, nil, tc.matchers, tc.shard, cachedSeriesHasher{shc}, nil, sl, true, b.meta.MinTime, b.meta.MaxTime, log.NewNopLogger(), false)
 			require.NoError(t, err, "Unexpected error for test case %d", i)
 			lset := lsetFromSeriesSet(t, ss)
 			require.Equalf(t, tc.expectedLabelSet, lset, "Wrong label set for test case %d", i)
 		}
 	})
 }
 
+func TestBlockSeries_IgnoreNativeHistograms(t *testing.T) {
+	newTestBucketBlock := prepareTestBlockWithBinaryReader(test.NewTB(t), func(t testing.TB, app storage.Appender) {
+		var err error
+		_, err = app.Append(0, labels.FromStrings("foo", "bar"), 10, 10)
+		require.NoError(t, err)
+		_, err = app.AppendHistogram(0, labels.FromStrings("foo", "bar"), 20, e2e.GenerateTestHistogram(1), nil)
+		require.NoError(t, err)
+		_, err = app.AppendHistogram(0, labels.FromStrings("foo", "bar"), 30, nil, e2e.GenerateTestFloatHistogram(1))
+		require.NoError(t, err)
+		require.NoError(t, app.Commit())
+	})
+
+	b := newTestBucketBlock()
+	b.shouldIgnoreNativeHistogramChunks = func() bool { return true }
+	b.indexCache = newInMemoryIndexCache(t)
+
+	cl := NewLimiter(math.MaxUint64, promauto.With(nil).NewCounter(prometheus.CounterOpts{Name: "test"}))
+	sl := NewLimiter(math.MaxUint64, promauto.With(nil).NewCounter(prometheus.CounterOpts{Name: "test"}))
+	shc := hashcache.NewSeriesHashCache(1 << 20).GetBlockCache(b.meta.ULID.String())
+
+	indexr := b.indexReader()
+
+	chunkr := b.chunkReader(context.Background())
+
+	chunksPool := pool.NewSafeSlabPool[byte](chunkBytesSlicePool, chunkBytesSlabSize)
+	defer chunksPool.Release()
+
+	matchers := []*labels.Matcher{
+		labels.MustNewMatcher(labels.MatchEqual, "foo", "bar"),
+	}
+
+	ss, _, err := blockSeries(context.Background(), indexr, chunkr, chunksPool, matchers, nil, cachedSeriesHasher{shc}, cl, sl, false, b.meta.MinTime, b.meta.MaxTime, log.NewNopLogger(), true)
+	require.NoError(t, err)
+
+	for ss.Next() {
+		_, chks := ss.At()
+		for _, chk := range chks {
+			require.Equal(t, storepb.Chunk_XOR, chk.Raw.Type)
```

Comment:
and to be on the safe side you can also assert that the chunk content isn't empty at least

Reply:
Good call - will do!